src/adapter/optimizations/item_lookup.rs

```diff
@@ -298,7 +319,18 @@ fn resolve_items_by_importable_path_field_value<'a>(
         .expect("crate's imports_index was never constructed")
         .get(path_components.as_slice())
     {
-        resolve_item_vertices(origin, items.iter().map(|(item, _)| item).copied())
+        resolve_item_vertices(
+            origin,
+            items
+                .iter()
+                .map(|(item, _)| item)
+                .copied()
+                .filter(move |item| {
+                    crate_vertex
+                        .pub_item_kind_index
+                        .contains(destination_type.clone(), item.id)
```

Comment:
See if we can avoid cloning the `Arc` on every iteration? Based on a quick glance, it feels like we should be able to move it into the closure without cloning on every call.